my-calculator/src/components/Calculator.js

```diff
@@ -0,0 +1,34 @@
+import React, {Component} from 'react';
+import Display from './Display';
+import KeyPad from './KeyPad';
+
+class Calculator extends Component{
+    state = {
+        displayValue: '0',
+        previousValue: null,
+        operation: null,
+        waitingForNewValue: false
+    }
+    handleOnClick(button) {
```

Comment:
The problem you have here is that you are creating the methods in es5, this will bring you a problem with the keyword `this`. for now i would suggest only creating the methods in es6 format, so instead of `handleOnClick (button){ do this }` you can write `handleOnClick = button => { do that}`
Which by the way should take in the event as the argument.